### PR TITLE
tests: build the initramfs + kernel snap for UC20 spread tests

### PR DESCRIPTION
**Pull request by anonymouse64**
Just opening this as a draft to see if it works on spread yet.

Also probably want @sergiocazzolato to review this when he's back next week before merging.

### CODE REVIEW

**Comment by bboozzoo** [comment]
@anonymouse64 i've included your patches to use the `test-snapd-ubuntu-core-initramfs` and pushed 2 fixes on top. If we can get @sergiocazzolato to update the host image to an UEFI enabled 20.04 we may use the tooling directly rather than though a snap.

File: tests/lib/prepare.sh [hunk 1]
```diff
@@ -656,13 +705,15 @@ EOF
     # mount fresh image and add all our SPREAD_PROJECT data
     kpartx -avs "$IMAGE_HOME/$IMAGE"
     # FIXME: hardcoded mapper location, parse from kpartx
+    devloop=$(losetup --list --noheadings | grep "$IMAGE_HOME/$IMAGE" | awk '{print $1}')
```
Comment (anonymouse64) [inline tests/lib/prepare.sh]: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1:

Comment (bboozzoo) [reply tests/lib/prepare.sh]: Landed in master as #8097

**Comment by anonymouse64** [comment]
#8094 will replace this

**Pull request closed without merging**